gateway/src/main/java/io/zeebe/gateway/impl/broker/cluster/BrokerTopologyManagerImpl.java

```diff
@@ -79,123 +72,112 @@ public BrokerClusterState getTopology() {
+      }
     }
-    final ActorFuture<ClientResponse> responseFuture =
-        output.sendRequest(endpoint, topologyRequest, Duration.ofSeconds(1));
-
-    refreshAttempt++;
-    lastRefreshTime = ActorClock.currentTimeMillis();
-    actor.runOnCompletion(responseFuture, this::handleResponse);
-    actor.runDelayed(MAX_REFRESH_INTERVAL_MILLIS, scheduleIdleRefresh());
   }
 
-  /** Only schedules topology refresh if there was no refresh attempt in the last ten seconds */
-  private Runnable scheduleIdleRefresh() {
-    final int currentAttempt = refreshAttempt;
+  private void updatePartitions(
+      String propertyName, String propertyValue, int brokerId, BrokerClusterStateImpl newTopology) {
+    final int partitionId = Integer.parseInt(propertyName.split("-")[1]);
+    newTopology.addPartitionIfAbsent(partitionId);
 
-    return () -> {
-      // if no topology refresh attempt was made in the meantime
-      if (currentAttempt == refreshAttempt) {
-        actor.run(this::refreshTopology);
-      }
-    };
+    if (RaftState.valueOf(propertyValue) == RaftState.LEADER) {
+      newTopology.setPartitionLeader(partitionId, brokerId);
+      LOG.info("Gateway: Added broker {} as leader of partition {}", brokerId, partitionId);
+    }
   }
 
-  private void handleResponse(final ClientResponse clientResponse, final Throwable t) {
-    if (t == null) {
-      final BrokerResponse<TopologyResponseDto> response =
-          topologyRequest.getResponse(clientResponse);
-      if (response.isResponse()) {
-        onNewTopology(response.getResponse());
-      } else {
-        failRefreshFutures(new RuntimeException("Failed to refresh topology: " + response));
-      }
-    } else {
-      failRefreshFutures(t);
+  private void updateBrokerClientAddress(
+      String clientApiAddress, int brokerId, BrokerClusterStateImpl newTopology) {
+    try {
+      final String socketAddress = objectMapper.readValue(clientApiAddress, String.class);
+      registerEndpoint.accept(brokerId, SocketAddress.from(socketAddress));
+      newTopology.setBrokerAddressIfPresent(brokerId, socketAddress);
+    } catch (IOException e) {
+      Log.error("Gateway: Invalid client address for broker {}:  ", Integer.toString(brokerId), e);
     }
   }
 
-  private void onNewTopology(final TopologyResponseDto topology) {
-    final BrokerClusterStateImpl newClusterState =
-        new BrokerClusterStateImpl(topology, registerEndpoint);
-    this.topology.set(newClusterState);
-    completeRefreshFutures(newClusterState);
-  }
+  private void updateClusterInfo(
+      String propertyName, String propertyValue, int brokerId, BrokerClusterStateImpl newTopology) {
 
-  private void completeRefreshFutures(final BrokerClusterStateImpl newClusterState) {
-    nextTopologyFutures.forEach(f -> f.complete(newClusterState));
-    nextTopologyFutures.clear();
-  }
+    try {
+      switch (propertyName) {
+        case "replicationFactor":
```

Comment:
Again we should use some type safe POJO instead of parsing every properties on its own